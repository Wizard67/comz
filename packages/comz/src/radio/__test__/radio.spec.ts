import { mount } from '@vue/test-utils'
import Radio from '../radio.vue'

describe('Radio.vue', () => {
  describe('props', () => {
    it('value', () => {
      const content = 'radio'

      const wrapper = mount(Radio, {
        props: {
          modelValue: '',
          value: content
        }
      })

      const element = wrapper.find('.cradio__field').element

      expect((element as HTMLInputElement).value).toBe(content)
      expect(wrapper.element).toMatchSnapshot()
    })

    it('modelValue', () => {
      const value = 'radio'

      const wrapper = mount(Radio, {
        props: {
          modelValue: value,
          value: value
        }
      })

      expect(wrapper.classes()).toContain('cradio--checked')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('disabled', () => {
      const wrapper = mount(Radio, {
        props: {
          modelValue: '',
          disabled: true
        }
      })

      expect(wrapper.classes()).toContain('cradio--disabled')
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('events', () => {
    it('update:modelValue', async () => {
      const value = 'radio'

      const wrapper = mount(Radio, {
        props: {
          modelValue: '',
          value: value
        }
      })

      await wrapper.find('.cradio__field').setValue(true)

      expect(wrapper.emitted()).toHaveProperty('update:modelValue')
      expect(wrapper.emitted()['update:modelValue'][0]).toEqual([value])
    })
  })

  describe('slots', () => {
    it('default', () => {
      const content = "text"

      const wrapper = mount(Radio, {
        props: {
          modelValue: ''
        },
        slots: {
          default: content
        }
      })

      expect(wrapper.text()).toContain(content)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
