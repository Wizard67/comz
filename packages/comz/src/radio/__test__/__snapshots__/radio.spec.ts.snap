// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio.vue props disabled 1`] = `
<label
  class="cradio cradio--disabled"
>
  <div
    class="cradio__button cradio__button--disabled"
  >
    ​
  </div>
  <input
    class="cradio__field"
    disabled=""
    type="radio"
    value="false"
  />
  
  
</label>
`;

exports[`Radio.vue props modelValue 1`] = `
<label
  class="cradio"
>
  <div
    class="cradio__button cradio__button--checked"
  >
    ​
  </div>
  <input
    class="cradio__field"
    type="radio"
    value="radio"
  />
  
  
</label>
`;

exports[`Radio.vue props value 1`] = `
<label
  class="cradio"
>
  <div
    class="cradio__button"
  >
    ​
  </div>
  <input
    class="cradio__field"
    type="radio"
    value="radio"
  />
  
  
</label>
`;

exports[`Radio.vue slots default 1`] = `
<label
  class="cradio"
>
  <div
    class="cradio__button"
  >
    ​
  </div>
  <input
    class="cradio__field"
    type="radio"
    value="false"
  />
  
  
  text
  
  
</label>
`;
